### ISSUE TITLE
Metabase 0.25.1 does not take in-depth database analysis flag into account

### ISSUE THREAD

**Issue by ruleb** [metabase/metabase#5681]
- Your databases: Postgres 9.5.4, Postgres 9.4.7
- Metabase version: 0.25.1
- Metabase hosting environment: Docker
- Metabase internal database: Postgres

We had been a while on 0.24.2, which runs fine, with in-depth database analysis enabled on all DBs.

We started noticing a change in load on our databases after upgrading to 0.25.1.
![screenshot 2017-08-04 09 42 05](https://user-images.githubusercontent.com/1880435/28959391-8d1e5868-78fa-11e7-8901-824c94fa3989.png)

Queries which are fired off to the databases are all distinct value searches for the top 301 rows for every column. This is deadly on large tables with a lot of columns.
```
SELECT $column AS $column
FROM $table
GROUP BY $column 
ORDER BY $column ASC LIMIT 301
```

After discovering the issue, we disabled in-depth database analysis, but the queries continued.
![screenshot 2017-08-04 09 43 48](https://user-images.githubusercontent.com/1880435/28959467-d8a23368-78fa-11e7-8e9d-6c0a7c37ce4d.png)

We restarted the metabase service to make sure it applied the database config changes, but the distinct queries continued.

After that we killed the metabase environment for a night and all database load stopped.

No dashboards/pulses where executed at the time of tests, no dashboards include these distinct queries.

Our databases are quite large, it needs to be possible to disable all of this behavior.

We have now rolled back to 0.24.2.

**Comment by Oberonhive** [comment]
Same issue. Metabase should select distinct values but last version select billions of rows on each field.

**Comment by salsakran** [comment]
Hey all -

We're in the middle of a fairly large refactor + rewrite of how we do this. It's in our top 3 priorities for this release. You can get excessive details in #5680 and #5679.

One way or another we're going to get to the bottom of these issues. 

Can you help us out by letting me know what's going on in your server logs at that time? We have a two different processes that are currently referred to as "in-depth" analysis that might be the root cause.

**Comment by HelmiRifai** [comment]
Same issue occurs with us, let me know if we can help somehow to gather more info.

**Comment by atharvai** [comment]
the issue I'm seeing or feature I'm looking for is posted here: http://discourse.metabase.com/t/sync-specific-queryable-tables/1841/1

**Comment by tomeli5n** [comment]
@salsakran Same issue here.
The CPU utilization reach to 100%, by mysql.
The query is 
```
-- Metabase
SELECT count(*) AS `count` FROM `saldos_por_tipo`
```
"Saldos_por_tipo" is an user definied function and the state is "Copying to tmp table".

- In-depth analisys flag is disabled.
- Your databases: Mysql 5.5.41
- Metabase version: 0.25.1
- Metabase hosting environment: Ubuntu 14.04 (jar)
- Metabase internal database: h2

**Comment by camsaul** [comment]
The in-depth flag should be respected again in 0.25.2, just released

**Comment by camsaul** [comment]
Leaving this open to make sure I resolve other issues related to field values queries and improve our test story here for 0.26.0

**Comment by camsaul** [comment]
This was fixed in 0.25.2 by #5710
